apps/prairielearn/elements/pl-checkbox/test_refactor.py

```diff
@@ -321,10 +321,40 @@ def build_element(
         == pl_checkbox.PartialCreditType.EACH_ANSWER
     )
 
-    with pytest.raises(ValueError, match="Unknown partial_credit_method"):
+    with pytest.raises(ValueError, match=r"partial-credit-method is deprecated.*"):
         pl_checkbox.get_partial_credit_mode(build_element(True, "INVALID"))
 
 
+def test_partial_credit_new() -> None:
+    """Test that internal enum conversion works correctly."""
+
+    def build_element(partial_credit_type: str) -> lxml.html.HtmlElement:
+        attrs = 'answers-name="test"'
+        attrs += f' partial-credit="{partial_credit_type}"'
+        return lxml.html.fragment_fromstring(f"<pl-checkbox {attrs}></pl-checkbox>")
+
+    assert (
+        pl_checkbox.get_partial_credit_mode(build_element("off"))
+        == pl_checkbox.PartialCreditType.OFF
+    )
+
+    assert (
+        pl_checkbox.get_partial_credit_mode(build_element("net-correct"))
+        == pl_checkbox.PartialCreditType.NET_CORRECT
+    )
+    assert (
+        pl_checkbox.get_partial_credit_mode(build_element("coverage"))
+        == pl_checkbox.PartialCreditType.COVERAGE
+    )
+    assert (
+        pl_checkbox.get_partial_credit_mode(build_element("each-answer"))
+        == pl_checkbox.PartialCreditType.EACH_ANSWER
+    )
+
+    with pytest.raises(ValueError, match=r"Invalid partial-credit-method.*"):
+        pl_checkbox.get_partial_credit_mode(build_element("other"))
```

Comment:
We have some code that uses `set(answers)` or `set(correct_answers)`. Can we add a test for cases in which there are duplicates?